lib/App/Sqitch/Command/revert.pm

```diff
@@ -134,10 +144,28 @@ sub execute {
     my $change = $self->modified
         ? $engine->planned_deployed_common_ancestor_id
         : $self->to_change // shift @{ $changes };
-    $self->warn(__x(
-        'Too many changes specified; reverting to "{change}"',
-        change => $change,
-    )) if @{ $changes };
+
+    if (!defined $change && $self->strict) {
+        hurl {
+            ident   => 'revert:strict',
+            message => __ 'Must specify a target revision in strict mode',
+            exitval => 1,
+        };
+    }
+    elsif (@{ $changes }) {
+        if ($self->strict) {
+            hurl {
+                ident   => 'revert:strict:excess',
+                message => __ 'Too many changes specified',
+                exitval => 1,
+            };
+        } else {
```

Comment:
Don't need the `else` here; `hurl` raises an exception.

Reply:
How's this?